### PR TITLE
Decorator to run upgrade test on perticualr satellite version

### LINKED ISSUE #25: Create a decorator/mark to run some tests only on particular sat version
There are some tests where the entity attribute data was not available in 6.1 but avaiable in 6.2

These tests fails during existence tests run as 6.2 value compares with 6.1 none.

The tests with such issue are as follows:
- [x] test_positive_smart_params_by_puppet_class
- [x] All Smart Variables tests
- [x] All Settings tests
- [x] test_positive_cvs_by_composite_views

### PR DESCRIPTION
**Pull request by jyejare**
closes #25

### CODE REVIEW

**Comment by jyejare** [comment]
The skip logs:

```
$ FROM_VERSION='6.1' py.test upgrade_tests/test_existance_relations/test_scparams.py -k test_positive_smart_params_by_puppet_class
=============================================================================================== test session starts ===============================================================================================
platform linux2 -- Python 2.7.10, pytest-3.0.7, py-1.4.33, pluggy-0.4.0
rootdir: /home/jitendrayejare2/Desktop/RedHat/sat6-upgrade/satellite6-upgrade, inifile:
plugins: xdist-1.15.0, cov-2.4.0
collected 6880 items 

upgrade_tests/test_existance_relations/test_scparams.py ssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss

============================================================================================== 5160 tests deselected ==============================================================================================
================================================================================== 1720 skipped, 5160 deselected in 4.79 seconds ==================================================================================

$ FROM_VERSION='6.1' py.test upgrade_tests/test_existance_relations/test_smartvariables.py
=============================================================================================== test session starts ===============================================================================================
platform linux2 -- Python 2.7.10, pytest-3.0.7, py-1.4.33, pluggy-0.4.0
rootdir: /home/jitendrayejare2/Desktop/RedHat/sat6-upgrade/satellite6-upgrade, inifile:
plugins: xdist-1.15.0, cov-2.4.0
collected 4 items 

upgrade_tests/test_existance_relations/test_smartvariables.py ssss

============================================================================================ 4 skipped in 0.14 seconds ============================================================================================


$ FROM_VERSION='6.1' py.test upgrade_tests/test_existance_relations/test_settings.py
=============================================================================================== test session starts ===============================================================================================
platform linux2 -- Python 2.7.10, pytest-3.0.7, py-1.4.33, pluggy-0.4.0
rootdir: /home/jitendrayejare2/Desktop/RedHat/sat6-upgrade/satellite6-upgrade, inifile:
plugins: xdist-1.15.0, cov-2.4.0
collected 3 items 

upgrade_tests/test_existance_relations/test_settings.py sss

============================================================================================ 3 skipped in 0.13 seconds ============================================================================================


$ FROM_VERSION='6.1' py.test upgrade_tests/test_existance_relations/test_contentviews.py -k test_positive_cvs_by_composite_views
=============================================================================================== test session starts ===============================================================================================
platform linux2 -- Python 2.7.10, pytest-3.0.7, py-1.4.33, pluggy-0.4.0
rootdir: /home/jitendrayejare2/Desktop/RedHat/sat6-upgrade/satellite6-upgrade, inifile:
plugins: xdist-1.15.0, cov-2.4.0
collected 20 items 

upgrade_tests/test_existance_relations/test_contentviews.py sssss

=============================================================================================== 15 tests deselected ===============================================================================================
==================================================================================== 5 skipped, 15 deselected in 0.14 seconds =====================================================================================
```

File: upgrade_tests/helpers/common.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+cur_ver = os.environ.get('FROM_VERSION')
+
+
+class VersionException(Exception):
+    """Version Exception if wrong satellite version provided"""
+    pass
+
+
+def run_to_upgrade(version):
+    """Decorator on test definition to run that test on given 'version' only.
```
Comment (renzon) [inline upgrade_tests/helpers/common.py]: Docstring states this is a decorator. Decorator which receive parameter, like this one, must define a function which will receive decorated function as parameter. Besides that I was not able to find such a function inside of the body of this decorator. So I ask: have you decorated some test function and tested decorator?

**Comment by jyejare** [comment]
@renzon , Yeah the logs are telling that its working perfectly. By 'Decorator' I mean this will be called as decorator, but I dont what we call for this type of functions, but this approach works 100 %

File: upgrade_tests/helpers/common.py (hunk 1 above)
Comment (renzon) [inline upgrade_tests/helpers/common.py]: My bad, 'pytest.mark.skipif' is receiveing decorated function

**Comment by jyejare** [comment]
@SatelliteQE/quality-engineers , Hey folks, I need one more ACK here. 

Please help.

**Pull request merged**